### PR TITLE
feat: add pipfile support to docker image

### PR DESCRIPTION
**Pull request by lwywoo**
- [x] Ready for review
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Add pipfile support to dockerfile

**Test Locally**

`
docker build -t pipfiletest -f Dockerfile.python-3 .
docker run -it  -e SNYK_TOKEN=<INSERT_TOKEN> -v $PWD:/project pipfiletest:latest -d test
`

https://github.com/snyk/snyk/issues/786

### CODE REVIEW

File: docker/docker-python-entrypoint.sh [hunk 1]
```diff
@@ -1,6 +1,12 @@
 #!/bin/bash
+if [ -f "${PROJECT_PATH}/requirements.txt" ]; then
+    virtualenv -p python snyk
+    source snyk/bin/activate
+    pip install -U -r "${PROJECT_PATH}/requirements.txt"
+elif [ -f "${PROJECT_PATH}/Pipfile" ]; then
+    cd "${PROJECT_PATH}"
```
[the remark this answers is not in the document]
Comment (lwywoo) [reply docker/docker-python-entrypoint.sh]: Initially tried to install with full path but was getting:
`
✔ Successfully created virtual environment!
Virtualenv location: /home/node/.local/share/virtualenvs/node-lHasYD0_
Creating a Pipfile for this project…
Installing /project/Pipfile…
WARNING: Invalid requirement, parse error at "'/project'"
✘ Installation Failed
Failed to run the process ...
Failed to test pip project`

Will serarch the docs to see if I can find a more efficent way to do this

[the remark this answers is not in the document]
Comment (adrukh) [reply docker/docker-python-entrypoint.sh]: Thanks! Don’t think this is a ‘no go’, just worth seeing if we can keep the simplicity

[the remark this answers is not in the document]
Comment (garethr) [reply docker/docker-python-entrypoint.sh]: To enter the directory, and then return from whence you came, you could replace this with:

```
pushd "${PROJECT_PATH}"
```

Then after running pipenv run `popd`

File: docker/docker-python-entrypoint.sh [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+    pipenv install Pipfile
```
Comment (garethr) [inline docker/docker-python-entrypoint.sh]: Using `pipenv install` here has some interesting side effects that I don't think are desirable here. Consider the case you have a Pipfile and a Pipfile.lock.

My understanding is that `pipenv install` will install all dependencies, and update `pipfile.lock` with the versions it used. So it's not checking what you've checked in.

`pipenv sync` will install the exact versions specified in pipfile.lock and I think is a better solution when you detect a lock file.

The logic I have for this in the experimental images I'm plating with is https://github.com/garethr/snyk-images/blob/master/docker-entrypoint.sh#L14-L35

Comment (lwywoo) [reply docker/docker-python-entrypoint.sh]: Thanks for the info, will update the PR :)

**Comment by dkontorovskyy** [comment]
Why not use glob to match python custom `requirements.txt` files?

**Comment by miiila** [comment]
@dkontorovskyy Would you match all .txt files? Or just `requirements`-something?

**Comment by dkontorovskyy** [comment]
All `*req*.txt` files

**Comment by miiila** [comment]
@dkontorovskyy I thought about it more and I'm probably for keeping it for `requirements.txt` only (or parametrize it somehow). Because we would need to check, if something matching *req*.txt and we would need find that particular name (I don't know, how to do it in shell), because we would need to provided the full name. Additionally - not sure what to do if there would be multiple *req*.txt files.

**Comment by garethr** [comment]
Note that the new, experimental, set of images supports these use cases out of the box.

https://github.com/snyk/snyk-images

* It correctly detects Pipfile and can test Pipfile baseed projects
* It supports an environment variable which can be used to override the defaults

More details in the README: https://github.com/snyk/snyk-images#running-bootstrap-commands

It may be worth trying that out.

**Comment by lili2311** [comment]
Will add this support in this PR: https://github.com/snyk/snyk/pull/925 please come over to review here instead @garethr @adrukh

**Pull request closed without merging**